eth-custodian/contracts/EthCustodian.sol

```diff
@@ -106,10 +108,11 @@ contract EthCustodian is ProofKeeper, AdminControlled {
 
     /// Withdraws the appropriate amount of ETH which is encoded in `proofData`
     function withdraw(
-        bytes calldata proofData, 
+        bytes calldata proofData,
         uint64 proofBlockHeight
     )
         external
+        nonReentrant
```

Comment:
This is a user exposed function that needs to be called each time to withdraw the funds. I think it is better not to add `modifier` flag given that it doesn't provide any more security.

> though could be useful for any future improvements.

Any future improvements should keep this contract as simple as possible (it is already not very simple) so I wouldn't expect any modification of this function. It is worth adding a comment to the function letting the developers know why is this function not susceptible to reentrance attacks.

Reply:
I agree with @mfornet : If the proof can’t be reproduced, so there is no need for adding the reentrancy modifier.